tests/test_register.py

```diff
@@ -154,30 +154,25 @@ def test_register_set_storage():
     assert register.storage == storage
 
 
+@pytest.mark.parametrize(
+    "dtype, reg_range, reg_type",
+    [
+        (REG_DTYPE.U8, (0, 100), int),
+        (REG_DTYPE.FLOAT, (0.0, 1.0), float),
+        (REG_DTYPE.S16, (-100, None), int),
+        (REG_DTYPE.U32, (None, 100), int),
+    ],
+)
 @pytest.mark.no_connection
-def test_register_range():
-    access = REG_ACCESS.RW
-
-    # custom range
-    dtype = REG_DTYPE.U8
-    range = (0, 100)
-    register = Register(dtype, access, reg_range=range)
-    assert type(register.range[0]) is int
-    assert type(register.range[1]) is int
-    assert register.range == range
-
-    # custom range float
-    dtype = REG_DTYPE.FLOAT
-    range = (1.11, 100.25)
-    register = Register(dtype, access, reg_range=range)
-    assert type(register.range[0]) is float
-    assert type(register.range[1]) is float
-    assert register.range == range
-
-    # default range
-    dtype = REG_DTYPE.U8
-    register = Register(dtype, access)
-    assert register.range == (dtypes_ranges[dtype]["min"], dtypes_ranges[dtype]["max"])
+def test_register_range(dtype, reg_range, reg_type):
+    register = Register(dtype, REG_ACCESS.RW, reg_range=reg_range)
+
+    assert type(register.range[0]) is reg_type
+    assert type(register.range[1]) is reg_type
+    if None in reg_range:
+        assert register.range == (dtypes_ranges[dtype]["min"], dtypes_ranges[dtype]["max"])
```

Comment:
Why, if exist a None in the range, all the range is ignored and the maximum and the minimum of the register type is set?

Reply:
Done